source/common/conn_pool/conn_pool_base.h

```diff
@@ -108,6 +108,7 @@ class ActiveClient : public LinkedObject<ActiveClient>,
   // (functionally) unlimited.
   // TODO: this could be moved to an optional to make it actually unlimited.
   uint32_t remaining_streams_;
+  uint32_t configured_stream_limit_;
```

Comment:
mind commenting?  there's a couple of configured limits (max streams, max concurrent)